src/Compilers/CSharp/Portable/Symbols/SubstitutedTypeParameterSymbol.cs

```diff
@@ -118,6 +118,6 @@ internal override TypeSymbol GetDeducedBaseType(ConsList<TypeParameterSymbol> in
             return _map.SubstituteType(_underlyingTypeParameter.GetDeducedBaseType(inProgress)).AsTypeSymbolOnly();
         }
 
-        private static readonly Func<TypeSymbolWithAnnotations, bool> s_isNotObjectFunc = type => type.SpecialType != SpecialType.System_Object;
+        private static readonly Func<TypeSymbolWithAnnotations, bool> s_isNotObjectFunc = type => type.SpecialType != SpecialType.System_Object || !type.IsAnnotated;
```

Comment:
>s_isNotObjectFunc [](start = 70, length = 17)

The name seems misleading now since it looks like the lambda represents "not `object?`". Consider renaming, or just inline the lambda in the one use above.

Reply:
> The name seems misleading now since it looks like the lambda represents "not object?". Consider renaming, or just inline the lambda in the one use above.

Will follow up on this in the next  PR

---
In reply to: [217497755](https://github.com/dotnet/roslyn/pull/29809#discussion_r217497755) [](ancestors = 217497755)